win32ss/user/user32/controls/appswitch.c

```diff
@@ -193,16 +199,17 @@ BOOL CALLBACK EnumerateCallback(HWND window, LPARAM lParam)
    }
 
    windowList[windowCount] = window;
-   iconList[windowCount] = CopyIcon(hIcon);
 
-   windowCount++;
+   for (cRetry = ICON_RETRY_COUNT; !hIcon && cRetry > 0; --cRetry)
+      Sleep(ICON_TIMEOUT / ICON_RETRY_COUNT);
```

Comment:
```suggestion
   for (cRetry = ICON_RETRY_COUNT; !hIcon && cRetry > 0; --cRetry)
   {
      Sleep(ICON_TIMEOUT / ICON_RETRY_COUNT);
   }
```

Reply:
I don't understand what this for-loop is for.
Actually I understand why you wrote it: you think that the hIcon could suddenly be initialized after some time.
However, it won't. When you call `SendMessageTimeoutW()` to retrieve the hIcon, it is this function itself that is waiting until the timeout expires/the hung condition happens, in which case it returns failure, and there is no chance you can later get a hIcon out of thin air.

Reply:
Deleted.